### PR TITLE
Fixes oscarlab/graphene#239 (shim_do_epoll_wait ignores user-supplied timeout)

### PR DESCRIPTION
**Pull request by dimakuv**
A trivial fix and a (less trivial) unit test for regression.

---
This change is [Reviewable](https://reviewable.io/reviews/oscarlab/graphene/249)

### CODE REVIEW

File: LibOS/shim/test/regression/epoll_wait_timeout.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/epoll.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+
+static int create_and_bind (char *port) {
```
Comment (donporter) [inline LibOS/shim/test/regression/epoll_wait_timeout.c]: Consider adding a comment about how to interpret the return value.  This function also seems to have some failure cases that do not print a clear error.

File: LibOS/shim/test/regression/epoll_wait_timeout.c [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+#define MAXEVENTS 64
+
+int main (int argc, char *argv[]) {
+    int sfd, s, n;
+    int efd;
+    struct epoll_event event;
+    struct epoll_event *events;
+
+    sfd = create_and_bind("8000");
```
Comment (donporter) [inline LibOS/shim/test/regression/epoll_wait_timeout.c]: You may need to make the port a command-line argument.  We have had a few issues with jenkins where, if the same ephemeral port is used in quick succession, the test will spuriously fail.

**Comment by donporter** [comment]
Ok to test

**Comment by dimakuv** [comment]
Updated the unit test as per @donporter comments.

**Comment by donporter** [comment]
I think we can merge this Sunday or when we get a third approval.

**Comment by donporter** [comment]
CI Failed on lat_proc exec, which is probably flaky.  Let's try again.

**Comment by donporter** [comment]
Test this please

**Comment by mkow** [comment]
I know I'm quite late, but I think it's worth fixing two things:
* Checking return value from `epoll_wait`
* I'm not sure if something lower than 1s wouldn't be better, is there any reason we wait here for so long? If not, I'd use something like 50ms.

What do you think?
We can do that in a separate PR if agree with these changes.

**Comment by donporter** [comment]
I"m ok with these changes in a separate PR

**Pull request merged**